edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter/IMessageConsumer.cs

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.MqttBrokerAdapter
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public interface IMessageConsumer
+    {
+        Task<bool> HandleAsync(MqttPublishInfo publishInfo);
+        void ProducerStopped();
```

Comment:
Can the producer restart? If so, how can it signal to the consumer it has restarted? If not, then maybe we can just use the dispose pattern.

Reply:
It cannot restart, however the consumers are injected services, so calling Dispose on them by the user class (the connector in this case) doesn't seem right to me - they don't own the instance. From the practical side, Dispose would be called by autofac, too - it doesn't cause problem (double dispose call), but to me it shows that that maybe Dispose() is not the right place to signal the consumer.